packages/front-end/components/Features/FeatureDiagnostics.tsx

```diff
@@ -0,0 +1,314 @@
+  results: Array<
+    FeatureEvalDiagnosticsQueryResponseRows[number] & { id: string }
+  > | null;
+  setResults: (
+    results: Array<
+      FeatureEvalDiagnosticsQueryResponseRows[number] & { id: string }
+    >,
+  ) => void;
+}) {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const { datasources, getDatasourceById } = useDefinitions();
+  const settings = useOrgSettings();
+  const { apiCall } = useAuth();
+
+  const validDatasources = useMemo(() => {
+    return datasources.filter((d) =>
+      isProjectListValidForProject(d.projects, feature.project),
+    );
+  }, [datasources, feature.project]);
+
+  const form = useForm({
+    defaultValues: {
+      ...getFeatureDatasourceDefaults(datasources, settings, feature.project),
+    },
+  });
+
+  const datasourceId = form.watch("datasourceId");
+  const datasource = datasourceId ? getDatasourceById(datasourceId) : null;
+
+  const datasourceHasFeatureUsageQuery =
+    datasource &&
+    datasource.settings.queries?.featureUsage &&
+    datasource.settings.queries.featureUsage.length > 0;
+
+  // Extract all unique keys from results (excluding 'id' which is added internally)
+  const columns = useMemo(() => {
+    if (results === null || results.length === 0) return [];
+    const keysSet = new Set<string>();
+    // Only iterate over the first row since all rows have the same structure
+    Object.keys(results[0]).forEach((key) => {
+      if (key !== "id" && key !== "timestamp" && key !== "feature_key") {
+        keysSet.add(key);
+      }
+    });
+
+    return Array.from(keysSet);
+  }, [results]);
+
+  const evalItems = useAddComputedFields(
+    results ?? [],
+    (row) => {
+      // Compute display values for all columns
+      const displayValues: Record<string, string> = {};
+      columns.forEach((key) => {
+        displayValues[key] = formatDisplayValue(row[key]);
+      });
+
+      return {
+        timestamp: format(getValidDate(row.timestamp), "PPpp"),
+        ...displayValues,
+      };
+    },
+    [results, columns],
+  );
+
+  const { items, pagination, SortableTH } = useSearch({
+    items: evalItems,
+    defaultSortField: "timestamp",
+    defaultSortDir: -1,
+    localStorageKey: "feature-diagnostics",
+    searchFields: ["timestamp", ...columns],
+    pageSize: 100,
+  });
+
+  const onRunFeatureUsageQuery = async () => {
+    setLoading(true);
+    setError(null);
+    try {
+      const results = await apiCall<FeatureEvaluationDiagnosticsQueryResults>(
+        "/query/feature-eval-diagnostic",
+        {
+          method: "POST",
+          body: JSON.stringify({
+            feature: feature.id,
+            datasourceId: form.watch("datasourceId"),
+          }),
+        },
+      );
+      if (results.rows) {
+        // add an id to each row based on the timestamp
+        const rowsWithId = results.rows.map((row) => ({
+          ...row,
+          id: row.timestamp ?? "",
```

Comment:
question: Why do we need the timestamp as an id? There's potential for conflict here.

Would using the array index work?